sdk/@launchdarkly/observability-python/ldobserve/__init__.py

```diff
@@ -1,7 +1,132 @@
+        *new_items: Variable number of items to add to the list.
+
+    Example:
+        >>> _extend_environment_list("MY_LIST", "item1", "item2")
+        # If MY_LIST was "existing_item", it will become "existing_item,item1,item2"
+    """
+    current_value = os.getenv(env_var_name, "")
+
+    # Split the current value by comma and strip whitespace
+    current_list = [item.strip() for item in current_value.split(",") if item.strip()]
+
+    # Add new items, avoiding duplicates
+    for item in new_items:
+        if item.strip() and item.strip() not in current_list:
+            current_list.append(item.strip())
+
+    # Join the list back into a comma-separated string
+    new_value = ",".join(current_list)
+
+    # Set the environment variable
+    os.environ[env_var_name] = new_value
+
+
+def _extend_experimental_resource_detectors(*detectors: str) -> None:
+    """
+    Extend the OTEL_EXPERIMENTAL_RESOURCE_DETECTORS environment variable with additional detectors.
+
+    This function reads the current value of OTEL_EXPERIMENTAL_RESOURCE_DETECTORS,
+    adds the provided detectors to the comma-separated list, and sets the
+    environment variable with the extended list.
+    """
+    _extend_environment_list(OTEL_EXPERIMENTAL_RESOURCE_DETECTORS, *detectors)
+
+
+def _extend_disabled_instrumentations(*instrumentations: str) -> None:
+    """
+    Extend the OTEL_PYTHON_DISABLED_INSTRUMENTATIONS environment variable with additional instrumentations.
+
+    This function reads the current value of OTEL_PYTHON_DISABLED_INSTRUMENTATIONS,
+    adds the provided instrumentations to the comma-separated list, and sets the
+    environment variable with the extended list.
+
+    Args:
+        *instrumentations: Variable number of instrumentation names to add to the disabled list.
+                          These should be the names of OpenTelemetry instrumentations to disable.
+
+    Example:
+        >>> extend_disabled_instrumentations("redis", "kafka")
+        # If OTEL_PYTHON_DISABLED_INSTRUMENTATIONS was "grpc_client",
+        # it will become "grpc_client,redis,kafka"
+    """
+    _extend_environment_list(OTEL_PYTHON_DISABLED_INSTRUMENTATIONS, *instrumentations)
+
+
+class ObservabilityPlugin(Plugin):
+    def __init__(self, config: Optional[ObservabilityConfig] = None):
+        processed_config = _ProcessedConfig(config or ObservabilityConfig())
+        self._config = processed_config
+        # Instruct auto-instrumentation to not instrument logging.
+        # We will either have already done it, or it is disabled.
+        _extend_disabled_instrumentations("logging")
+
+        if self._config.disabled_instrumentations:
+            _extend_disabled_instrumentations(*self._config.disabled_instrumentations)
+
+        # If the OTEL_EXPERIMENTAL_RESOURCE_DETECTORS environment variable is not set, then we will use the config.
+        if os.getenv(OTEL_EXPERIMENTAL_RESOURCE_DETECTORS, None) is None:
```

Comment:
Default is `None`. Side question, do we care if this is set but empty?

```suggestion
        if os.getenv(OTEL_EXPERIMENTAL_RESOURCE_DETECTORS) is None:
```

Reply:
I was thinking if someone attempted to set it, then we would leave it alone. But I guess someone may attempt to set it empty instead of unsetting it.